### PR TITLE
[AV-77843] Prepare Release Documentation for v1.2.1

### PR DESCRIPTION
**Pull request by aniket-Kumar-c**
## Jira

* AV-77843

## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

### How was this change tested and do you have evidence? _**(REQUIRED: Select at least 1)**_

- [ ] Manually tested
- [ ] Unit tested
- [ ] Acceptance tested
- [x] Unable to test / will not test (Please provide comments in section below)

### Testing

<details open>
  <summary>Testing</summary>
  
</details>

## Required Checklist:

- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if required)
- [x] I have run make fmt and formatted my code

## Further comments

### CODE REVIEW

File: docs/guides/1.2.1-upgrade-guide.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+Couchbase Capella Provider 1.2.0: Upgrade and Information Guide
+---
+
+# Couchbase Capella Provider 1.2.1: Upgrade and Information Guide
+
+## Bug Fixed
+
+* Export capella audit logs with [`couchbase-capella_audit_log_export`](https://registry.terraform.io/providers/couchbasecloud/couchbase-capella/latest/docs/resources/audit_log_export) was failing with the v1.2.0 release, which has been rectified in the v1.2.1 release.
+
+## Changes
```
Comment (RichardSmedley) [inline docs/guides/1.2.1-upgrade-guide.md:15]: Should there also be an update to https://github.com/couchbasecloud/terraform-provider-couchbase-capella/blob/main/CHANGELOG.md ?

File: docs/guides/1.2.1-upgrade-guide.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+layout: "couchbase-capella"
+page_title: "Couchbase Capella Provider 1.2.0: Upgrade and Information Guide"
+sidebar_current: "docs-couchbase-capella-guides-120-upgrade-guide"
+description: |-
+Couchbase Capella Provider 1.2.0: Upgrade and Information Guide
+---
+
+# Couchbase Capella Provider 1.2.1: Upgrade and Information Guide
+
+## Bug Fixed
```
Comment (rajagp) [inline docs/guides/1.2.1-upgrade-guide.md:11]:
```suggestion
## Bug Fixes
```

File: docs/guides/1.2.1-upgrade-guide.md (hunk 1 above)
Comment (aniket-Kumar-c) [reply docs/guides/1.2.1-upgrade-guide.md:15]: We generally create changelog pr, after release documentation get merged

**Review by a-atri — APPROVED** [review]
Good to merge

**Review by Lagher0 — APPROVED** [review]

**Review by SaicharanCB — APPROVED** [review]
+1

**Pull request merged**